internal/service/advancedcluster/data_source.go

```diff
@@ -52,25 +54,43 @@ func (d *ds) readCluster(ctx context.Context, diags *diag.Diagnostics, modelDS *
 	if flexClusterResp == nil && clusterResp == nil {
 		return nil
 	}
+	var result *TFModelDS
 	if flexClusterResp != nil {
-		modelOut := NewTFModelFlex(ctx, diags, flexClusterResp, nil)
-		if diags.HasError() {
-			return nil
-		}
-		return conversion.CopyModel[TFModelDS](modelOut)
+		result = convertFlexClusterToDS(ctx, diags, flexClusterResp)
+	} else {
+		result = convertBasicClusterToDS(ctx, diags, d.Client, clusterResp)
+	}
+	if result != nil {
+		result.UseEffectiveFields = modelDS.UseEffectiveFields
+	}
```

Comment:
```suggestion
	result.UseEffectiveFields = modelDS.UseEffectiveFields
```
Why not always align resulting state with the config value?

Reply:
if result is nil (because some error) it would panic if trying to update its field